### ISSUE TITLE
Gradle daemon still runs after closing vscode

### ISSUE THREAD

**Issue by ttc0419** [microsoft/vscode-gradle#225]
Extension Name: vscode-gradle
Extension Version: 2.4.10
OS Version: Linux 5.5.8-1-ck-skylake
VSCode version: 1.43.0-1

**Describe the bug**
After opening the project two processes are created:
1. ```java -classpath /home/william/.vscode/extensions/richardwillis.vscode-gradle-2.4.13/lib/gradle-tasks-server-all.jar com.github.badsyntax.gradletasks.Application 37391```
2. ```/usr/lib/jvm/java-13-jdk/bin/java --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.invoke=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp /home/william/.gradle/wrapper/dists/gradle-6.2.2-bin/byfcpklxknejjsdmgq0rkga/gradle-6.2.2/lib/gradle-launcher-6.2.2.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 6.2.2```

But the latter is still running after exiting vscode.

**To Reproduce**
1. Open a gradle project in vscode
2. Close it

**Expected behavior**
Exit without any process left.

**Output from "Gradle Tasks"**
None.

**Does the bug still exist if you disable all other extensions?**
Yes

**Additional context**
None

**Comment by badsyntax** [comment]
Hi @Galaxy0419, thanks for this report. How are you verifying the process still exists after closing vscode?

**Comment by ttc0419** [comment]
@badsyntax ```ps aux | grep java```

**Comment by badsyntax** [comment]
Thanks. I haven't looked into this but as the gradle server process is managed by vscode (it's just another vscode task) I would expect vscode to handle shutting down the process. I will look into this when I got some time. Sorry for the delay.

**Comment by badsyntax** [comment]
This process is "internal" to gradle and should not be manually stopped as it's basically a shared process that could be used by other processes outside of this extension.

See related issues: 

* https://github.com/gradle/gradle/issues/5998
* https://github.com/gradle/gradle/issues/8219

As [mentioned here](https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:why_is_there_more_than_one_daemon_process_on_my_machine):

> Gradle will kill any Daemon that has been idle for 3 hours or more, so you don’t have to worry about cleaning them up manually.

I can potentially consider to lower the idle time (using [daemonMaxIdleTime](https://github.com/gradle/gradle/blob/8fa05470b9ce7ecbf35b2a8ef029019f5f0894df/subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/DefaultGradleConnector.java#L116)) but that might result in unexpected side-effects. 

So I don't think there's anything I can do here, as it appears to be working as designed.

I'll leave this issue open to remind me to add some info about this to the docs.

**Comment by badsyntax** [comment]
In the next version you will be able to see running daemons and stop them:

<img width="1552" alt="Screenshot 2020-05-31 at 19 02 31" src="https://user-images.githubusercontent.com/102141/83359300-4f94b880-a371-11ea-856f-dd8863529df2.png">

I have also updated the README describing the daemon processes. Those changes will be  released with https://github.com/badsyntax/vscode-gradle/pull/433.

**Comment by badsyntax** [comment]
Released with https://github.com/badsyntax/vscode-gradle/releases/tag/3.1.0

**Comment by RobMaskell** [comment]
@badsyntax  I seem to have more of these than I need, they don't seem to be clearing themselves after 3 hours
```
ps -eo uid,pid,ppid,lstart,cmd | grep [j]ava
 1000    5273    1696 Thu Aug 27 13:52:41 2020 java -classpath /home/robm/.vscode/extensions/richardwillis.vscode-gradle-3.4.2/lib/gradle-server.jar com.github.badsyntax.gradle.GradleServer 36101
 1000    6764    1696 Thu Aug 27 15:19:00 2020 java -classpath /home/robm/.vscode/extensions/richardwillis.vscode-gradle-3.4.2/lib/gradle-server.jar com.github.badsyntax.gradle.GradleServer 44887
 1000   10575    1696 Thu Aug 27 17:15:48 2020 java -classpath /home/robm/.vscode/extensions/richardwillis.vscode-gradle-3.4.2/lib/gradle-server.jar com.github.badsyntax.gradle.GradleServer 40057
 1000   15168    1696 Thu Aug 27 18:07:00 2020 java -classpath /home/robm/.vscode/extensions/richardwillis.vscode-gradle-3.4.2/lib/gradle-server.jar com.github.badsyntax.gradle.GradleServer 36817
 1000   24019    1696 Thu Aug 27 22:09:32 2020 java -classpath /home/robm/.vscode/extensions/richardwillis.vscode-gradle-3.4.2/lib/gradle-server.jar com.github.badsyntax.gradle.GradleServer 33717
```
This is on the latest vscode and all up to date, I suspect I get a new one everytime I start vscode, last time I started I looked that the running Daemons as per your comment above about the new feature and only saw one

**Comment by badsyntax** [comment]
Thanks for letting me know. To state the obvious those aren't the daemons. This could be a result of a recent change to how the server process is managed. I'll verify, then add a fix with tests. I'll also update the README to add more explicit info on the java processes.

**Comment by RobMaskell** [comment]
Would it help if I created a new issue for tracking?

**Comment by badsyntax** [comment]
Yes please, with clear reproducible steps, as I wasn't able to replicate this issue on MacOS. Thanks.

**Comment by RobMaskell** [comment]
Opened https://github.com/badsyntax/vscode-gradle/issues/684 for this

**Comment by badsyntax** [comment]
@Galaxy0419 the fix i added for https://github.com/badsyntax/vscode-gradle/issues/684 appears to now kill the daemons when closing vscode. It'd be useful if you could test this on your end. I added the fix with [3.4.4](https://github.com/badsyntax/vscode-gradle/releases/tag/3.4.4)
